Added the possibility to create directories

rvmourik opened the pull request.

rvmourik committed:
Added the possibility to create directories

frankdejonge commented on src/RackspaceAdapter.php:
```diff
@@ -162,7 +162,23 @@ public function deleteDir($dirname)
+            $headers['Contents'] = 'None';
+        }
+
+        if (!key_exists('Content-Type', $headers)) {
```
Shouldn't this always be `application/directory`?

frankdejonge commented on src/RackspaceAdapter.php:
```diff
@@ -162,7 +162,23 @@ public function deleteDir($dirname)
+        }
+
+        if (!key_exists('Contents', $headers)) {
+            $headers['Contents'] = 'None';
```
Shouldn't this always be `None`?

rvmourik replied:
I wanted it to be dynamic, let's say the headers are already send through the config (for some reason). Maybe a implementation based on RackSpace but with slidely different values, one would be able to pass this through the config without being worried it's overwritten with 'None', the same goes for 'application/directory' in the Content-Type.

frankdejonge replied:
Alternatively, shouldn't it just have no content at all? Or is this in the docs as well?

frankdejonge commented:
Travis didn't run yet, however I have the feeling it's not going to pass since the tests have not been corrected to account for this case.

rvmourik commented:
Okay, i'm not very familiar with Travis and Unit testing (yet), so is this something that needs work from my end or yours?

frankdejonge commented:
@rvmourik the test case needs to be updated which is located here: https://github.com/thephpleague/flysystem-rackspace/blob/master/tests/RackspaceAdapterTests.php#L158

rvmourik committed:
Change testCreateDir to except a boolean instead of an array

rvmourik committed:
Remove Content: None header because it's not needed

rvmourik committed:
Added the second parameter to createDir

rvmourik committed:
Update RackspaceAdapterTests.php

rvmourik commented:
@frankdejonge I removed the Contents None (our Cloud support department gave this as something that needed to be present, but just tested it, and it works without). I also changed the test to an assertTrue, but it still fails.

frankdejonge commented on src/RackspaceAdapter.php:
```diff
@@ -162,7 +162,19 @@ public function deleteDir($dirname)
+
+        if ($config->has('headers')) {
+            $headers = $config->get('headers');
+        }
```
this can be changed to:

``` php
$headers = $config->get('headers', []);
```

rvmourik replied:
done

rvmourik committed:
Update RackspaceAdapter.php

rvmourik commented:
@frankdejonge can you tell me what needs to be done about the test in order to make it pass? Thanks.

frankdejonge commented:
@rvmourik I've merged and corrected the PR. Thanks!

rvmourik commented:
@frankdejonge thanks!
